src/middlewares/AuthValidator.js

```diff
@@ -0,0 +1,68 @@
+
+/**
+  * @param {object} request - request object
+  * @param {object} response - response object
+  * @param {next} next - calls next middleware
+  *
+  * @returns {object} - return response message
+  */
+const validateSignup = (request, response, next) => {
+  request
+    .checkBody('firstName')
+    .notEmpty()
+    .withMessage('First name field is required')
+    .trim()
+    .isLength({ min: 2, max: 15 })
+    .withMessage('First name should be between 2 to 15 characters')
+    .isAlpha()
+    .withMessage('First name should only contain alphabets')
+    .customSanitizer(firstName => firstName.toLowerCase());
+
+  request
+    .checkBody('lastName')
+    .notEmpty()
+    .withMessage('Last name field is required')
+    .trim()
+    .isLength({ min: 2, max: 15 })
+    .withMessage('Last name should be between 2 to 15 characters')
+    .isAlpha()
+    .withMessage('Last name should only contain alphabets')
+    .customSanitizer(lastName => lastName.toLowerCase());
+
+  request
+    .checkBody('userName')
+    .notEmpty()
+    .withMessage('Username field is required')
+    .trim()
+    .isLength({ min: 2, max: 20 })
+    .withMessage('Username should be between 2 to 20 characters');
+
+  request
+    .checkBody('email')
+    .notEmpty()
+    .withMessage('Email field is required')
+    .trim()
+    .isEmail()
+    .withMessage('Enter a valid email address')
+    .customSanitizer(email => email.toLowerCase());
+
+  request
+    .checkBody('password')
+    .notEmpty()
+    .withMessage('Password field is required')
+    .trim()
+    .isLength({ min: 8, max: 15 })
+    .withMessage('Password should be between 8 to 15 characters')
+    .matches(/[aA-zZ]+[0-9]+/)
+    .withMessage('Password must be alphanumeric characters');
+
+  const errors = request.validationErrors();
```

Comment:
I don't like this file. It needs to be refactored to have one validation middleware that validates all routes based on the different field validators.